linker/Mono.Linker/Annotations.cs

```diff
@@ -349,11 +349,40 @@ public void Pop ()
 			dependency_stack.Pop ();
 		}
 
+		bool IsNestedInPrivate (TypeDefinition td)
+		{
+			if (td.BaseType == null)
+				return false;
+
+			while (td.DeclaringType != null) {
+				td = td.DeclaringType;
+				if (td.IsPublic)
+					return false;
+			}
+
+			return true;
+		}
+
 		string TokenString (object o)
 		{
 			if (o == null)
 				return "N:null";
 
+			if (o is TypeReference) {
```

Comment:
`o is TypeReference t` will give you `t` immediately

Reply:
I knew there's faster way, forgot to check it before pushing. Thanks!